sdk/trace/provider_test.go

```diff
@@ -94,3 +100,21 @@ func TestSchemaURL(t *testing.T) {
 	tracerStruct := tracerIface.(*tracer)
 	assert.EqualValues(t, schemaURL, tracerStruct.instrumentationLibrary.SchemaURL)
 }
+
+func TestEnsureValidTracerProviderConfig(t *testing.T) {
+	envStore := ottest.NewEnvStore()
+	defer func() {
+		require.NoError(t, envStore.Restore())
+	}()
+	envStore.Record(env.SpanAttributesCountKey)
+	envStore.Record(env.SpanEventCountKey)
+	envStore.Record(env.SpanLinkCountKey)
+	require.NoError(t, os.Setenv(env.SpanEventCountKey, "111"))
```

Comment:
A test with negative values provided here would be good, to ensure that the defaults are correctly used.